hemework25-datestr.py

```diff
@@ -0,0 +1,160 @@
+#TASK1
+from functools import wraps
+def check_full(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_full():
+            raise ValueError("Storage is full.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+def check_empty(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+class BaseSequence:
+    """A base sequence class."""
+    def __init__(self, initial_items: list, capacity=10):
+        self._storage = initial_items
+        self._capacity = capacity
+
+    @check_empty
+    def put(self, element):
+        """Put an element into the sequence."""
+        if self.is_full():
```

Comment:
I would expect you to decorate the `put` method with the `check_full` decorator instead of `check_empty` - this way you wouldn't need this check here